File: seminar_paper/small_experiment_buffer_overflow/generic_tasks/chat_gpt_response.c
#include <stdio.h>
#include <stdlib.h>

#define BUFFER_SIZE 10

void process_elements(int *buffer, int size) {
    int index;

    printf("Enter an index (0-%d): ", size - 1);
    if (scanf("%d", &index) != 1) {
        printf("Invalid input.\n");
        return;
    }

    // ✅ Bounds checking
    if (index < 0 || index >= size) {
        printf("Index out of bounds.\n");
        return;
    }

    // Safe access
    printf("Selected element: %d\n", buffer[index]);

    // Example: use the selected element later
    int result = buffer[index] * 2;
    printf("Processed result: %d\n", result);
}

int main() {
    int buffer[BUFFER_SIZE] = {0, 1, 2, 3, 4, 5, 6, 7, 8, 9};

    process_elements(buffer, BUFFER_SIZE);

    return 0;
}

File: seminar_paper/small_experiment_buffer_overflow/generic_tasks/chat_gpt_response.py
def select_from_buffer(buffer):
    """
    Prompts user to select indices from buffer and returns selected elements.
    """
    print(f"Buffer has {len(buffer)} elements (0 to {len(buffer)-1})")

    user_input = input("Enter indices (comma-separated): ")

    try:
        indices = [int(i.strip()) for i in user_input.split(",")]
    except ValueError:
        raise ValueError("Invalid input: please enter integers only.")

    # Validate indices
    selected = []
    for i in indices:
        if 0 <= i < len(buffer):
            selected.append(buffer[i])
        else:
            raise IndexError(f"Index {i} out of range.")

    return selected


# Example usage
buffer = ["apple", "banana", "cherry", "date"]
result = select_from_buffer(buffer)
print("Selected:", result)